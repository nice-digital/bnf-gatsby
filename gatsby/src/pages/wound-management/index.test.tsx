import { render, screen, within, waitFor } from "@testing-library/react";
import React from "react";

import WoundManagementIndexPage, {
	type WoundManagementIndexPageProps,
} from "./";

const props: WoundManagementIndexPageProps = {
	data: {
		bnfWoundManagementIntroduction: {
			title: "Wound management products and elasticated garments",
			sections: [
				{
					title: "Introduction",
					slug: "introduction",
					content:
						" <p>The correct dressing for wound management depends not only on the type of wound but also on the stage of the healing process",
					order: 1,
				},
			],
		},
		allBnfWoundManagementTaxonomy: {
			taxonomies: [
				{
					title: "Basic wound contact dressings",
					slug: "basic-wound-contact-dressings",
				},
				{
					title: "Advanced wound dressings",
					slug: "advanced-wound-dressings",
				},
			],
		},
	},
};

describe("Medical devices index page", () => {
	it("should render the page title with the expected text", async () => {
		render(<WoundManagementIndexPage {...props} />);
		await waitFor(() => {
			expect(document.title).toStartWith(
				"Wound management products and elasticated garments"
			);
		});
	});

	it("should render meta description", async () => {
		render(<WoundManagementIndexPage {...props} />);
		await waitFor(() => {
			// eslint-disable-next-line testing-library/no-node-access
			expect(document.querySelector("meta[name=description]")).toHaveAttribute(
				"content",
				"Browse wound management products and elasticated garments, by type."
			);
		});
	});

	it("should match page snapshot", () => {
		render(<WoundManagementIndexPage {...props} />);
		expect(screen.getByRole("main")).toMatchSnapshot();
	});

	it("should add content start skip link target id to page header", () => {
		render(<WoundManagementIndexPage {...props} />);
		const heading1 = screen.getByRole("heading", {
			level: 1,
		});
		// eslint-disable-next-line testing-library/no-node-access
		expect(heading1.parentElement).toHaveProperty("id", "content-start");
	});

	it("should render the page heading with the expected text", () => {
		render(<WoundManagementIndexPage {...props} />);
		expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent(
			"Wound management products and elasticated garments"
		);
	});

	it("should render a navigation list of sub sections", () => {
		render(<WoundManagementIndexPage {...props} />);
		expect(
			screen.getByRole("navigation", {
				name: "Wound management product pages",
			})
		).toBeInTheDocument();
	});

	it("should render correct number of sections in the sub nav", () => {
		render(<WoundManagementIndexPage {...props} />);
		const subNav = screen.getByRole("navigation", {
			name: "Wound management product pages",
		});
		expect(within(subNav).getAllByRole("listitem")).toHaveLength(
			props.data.allBnfWoundManagementTaxonomy.taxonomies.length
		);
	});
});
