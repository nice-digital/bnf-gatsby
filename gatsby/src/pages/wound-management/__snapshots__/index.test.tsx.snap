// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wound management index page should match page snapshot 1`] = `
<main>
  <div
    class="container"
  >
    <nav
      aria-label="Breadcrumbs"
      role="navigation"
    >
      <ol
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="https://www.nice.org.uk/"
          >
            NICE
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/"
          >
            BNF
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <span>
            Wound management products and elasticated garments
          </span>
        </li>
      </ol>
    </nav>
    <script
      type="application/ld+json"
    >
      {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "https://www.nice.org.uk/",
        "name": "NICE"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/",
        "name": "BNF"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "name": "Wound management products and elasticated garments"
      }
    }
  ]
}
    </script>
    <div
      class="page-header"
      id="content-start"
    >
      <h1
        class="page-header__heading"
      >
        Wound management products and elasticated garments
      </h1>
    </div>
    <div
      class="grid grid--loose"
      data-testid="body"
    >
      <div
        class="hide-print"
        data-g="12 md:4 lg:3"
      >
        <nav
          aria-label="Wound management product pages"
          class="stacked-nav"
        >
          <p
            class="stacked-nav__root"
          >
            <a
              aria-current="false"
              href="/wound-management/"
            >
              Wound management products
            </a>
          </p>
          <ul
            class="stacked-nav__list"
          >
            <li
              class="stacked-nav__list-item"
            >
              <a
                aria-current="false"
                href="/wound-management/basic-wound-contact-dressings"
              >
                <span
                  class="stacked-nav__content-wrapper"
                >
                  <span>
                    Basic wound contact dressings
                  </span>
                </span>
              </a>
            </li>
            <li
              class="stacked-nav__list-item"
            >
              <a
                aria-current="false"
                href="/wound-management/advanced-wound-dressings"
              >
                <span
                  class="stacked-nav__content-wrapper"
                >
                  <span>
                    Advanced wound dressings
                  </span>
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        data-g="12 md:8 lg:9"
      >
        <nav
          aria-labelledby="navigate-to-section"
          class="nav"
        >
          <h2
            class="heading"
            id="navigate-to-section"
          >
            Navigate to section
          </h2>
          <ol
            aria-label="Jump links to sections on this page"
            class="linkList"
          >
            <li>
              <a
                href="#wound-type--154124684"
              >
                Pink (epitheliasing)
              </a>
            </li>
            <li>
              <a
                href="#wound-type-1889862293"
              >
                Red (granulating)
              </a>
            </li>
            <li>
              <a
                href="#wound-type-102222233"
              >
                Yellow (sloughy)(granulating)
              </a>
            </li>
            <li>
              <a
                href="#wound-type-530708321"
              >
                Black (Necrotic/ Eschar)
              </a>
            </li>
            <li>
              <a
                href="#wound-type-592109193"
              >
                Wounds with signs of infection
              </a>
            </li>
          </ol>
        </nav>
        <section
          aria-labelledby="introduction"
          class="woundManagementIndex"
        >
          <h2
            class="firstHeading"
            id="introduction"
          >
            Introduction
          </h2>
          <div>
             
            <p>
              The correct dressing for wound management depends not only on the type of wound but also on the stage of the healing process
            </p>
          </div>
        </section>
      </div>
    </div>
  </div>
</main>
`;
